### ISSUE TITLE
Email Images use Incorrect Src Path

### ISSUE THREAD

**Issue by jeremyevans6** [sharetribe/sharetribe#4154]
All images in emails (invites, confirmation, transactions, etc) are built with a broken path. All lack the domain name. 

##### Sharetribe Version:
719e5fa5ab56d476d5ae5c07f26f8508a7d26e70
##### Environment:
Ubuntu 20.04
Node v10.15.3
NPM 6.4.1
##### Server mode:
Production
##### Description:
I'm getting paths like `/system/wide_logos/1/header/logo-01.png?1593402420` in the src. Naturally, these images are broken since there's no domain. 

[There's a thread about this exact problem on the community forum.](https://www.sharetribe.com/community/t/broken-image-in-newsletter/1303/8)

This solution is dated (three years old), so I'm not sure if the `_communitylisting-update-v2.html.haml` existed at that time. I am using the new layouts, so that's one place where I'll likely be hardcoding my domain. The problems I have with this approach are that there are many diverse areas in which I'll need to add a domain and that it's clearly not intended to be this way. I'm hoping I'm missing something obvious.

##### Steps To Reproduce:
New install of ShareTribe
:local asset configuration
asset_host: mydomain.com (blank also fails to produce image paths)
Successful smtp and sendmail email set-ups (emails are sent and received without errors)

##### Additional information
I attempted manually adding my domain to the domains and community tables of the database. It corrected some links which were pointing toward the default subdomain installation but did not add the domain to file paths.

I looked at some of these email .yml's, finding that there's no special treatment to add the domain name. I'm not deeply familiar with Ruby on Rails, but it doesn't seem like img_tag() has any post-processing for the src that adds the domain name. The img_tag for the logo in the email headers, for example, looks like this in `app/views/email_design/email.html.haml`: 
`= image_tag('logos/full/default.png', alt: community.full_name(I18n.locale), style: 'margin: 0 auto;')`

I can understand how these paths are adequate for local assets, but how does this work for remote assets or emailed images? Since I'm the first to have this problem here, there's clearly something I'm missing.

**Comment by jeremyevans6** [comment]
My hack around this was to append the domain to the following files:
`
app\views\layouts\email-v2.html.haml
\app\views\community_mailer\_communitylisting-update-v2.html.haml
`

I'm only touching the new layout emails, since I'm using those, and the latter file has two image_tag lines to edit. I did in fact look through every email file (for v2) and these were the only ones with images.

What I did to add the domains dynamically was prepend a bit any image_tag src. This is how each image_tag src field looked when I was done:
`"https://#{@url_params[:host]}#{PREEXISTING IMG SRC}"`

This worked, with images displaying properly in emails. I can't shake the feeling I've done something naughty, but I suppose I'll deal with that when I update.

**Comment by bladealslayer** [comment]
@jeremyevans6 try setting your `user_asset_host` configuration option to the same URL as your `asset_host`, except having the protocol scheme as prefix. E.g. `https://www.example.com`.

**Comment by jeremyevans6** [comment]
I'm afraid that had no effect. The resulting path still lacked the domain name.

**Comment by bladealslayer** [comment]
That's weird, because setting the asset host should affect the image URLs, set by this line of code:

https://github.com/sharetribe/sharetribe/blob/master/config/application.rb#L176

When you set `user_asset_host`, could you check how the listing image links look in the rendered HTML on regular pages (not emails)? Are they with FQDN or only contain path without hostname?

**Comment by jeremyevans6** [comment]
Thanks for helping, @bladealslayer. You're spot on.

Without `asset_host` or `user_asset_host`, my images use relative paths. With `asset_host: mydomainname.com` and `user_asset_host: https://mydomainname.com`, I can confirm my listing images use an absolute filepath with the correct FQDN. 

The email images actually do work with this configuration! I'm rather surprised because, admittedly, I did a sloppy test last time: I left my tweak adding the domain name and expected a doubled FQDN in the path if it worked. Worse yet, I used Gmail for the test, which modifies sent images and puts them in its own cloud. I think Gmail edited out this doubled path, showing me the same pathname as before the new settings.

Looking at the raw email, I can see that your combination of settings does in fact correct this issue. Thank you! I'm back on the master branch (almost).
